flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/literals.scala

```diff
@@ -49,10 +50,51 @@ object Literal {
     case sqlTime: Time => Literal(sqlTime, SqlTimeTypeInfo.TIME)
     case sqlTimestamp: Timestamp => Literal(sqlTimestamp, SqlTimeTypeInfo.TIMESTAMP)
   }
+
+  private[flink] def apply(rexNode: RexLiteral): Literal = {
+    val literalType = FlinkTypeFactory.toTypeInfo(rexNode.getType)
+
+    val literalValue = literalType match {
+      // Chrono use cases.  We're force-adjusting the UTC-based epoch timestamps to a new
+      // timestamp such that we get the same year/month/hour/day field values in the query's
+      // timezone (UTC)
+      case _@SqlTimeTypeInfo.DATE =>
+        val rexValue = rexNode.getValueAs(classOf[DateString])
+        val adjustedCal = adjustCalendar(rexValue.toCalendar, TimeZone.getDefault)
+        new Date(adjustedCal.getTimeInMillis)
```

Comment:
This can be simplified to `new Date(rexValue.toString)`.

Reply:
Unfortunately that constructor is deprecated in Java 8, which is why I avoided using it.

Reply:
Do you want me to use the deprecated constructor or leave this as-is?